test/command_callback/test_cs_mcsc_command_callbacks.vader

```diff
@@ -0,0 +1,120 @@
+Before:
+  Save g:ale_cs_mcsc_options
+  Save g:ale_cs_mcsc_source
+  Save g:ale_cs_mcsc_assembly_path
+  Save g:ale_cs_mcsc_assemblies
+
+  unlet! g:ale_cs_mcsc_options
+  unlet! g:ale_cs_mcsc_source
+  unlet! g:ale_cs_mcsc_assembly_path
+  unlet! g:ale_cs_mcsc_assemblies
+
+  let g:temppath = fnamemodify(tempname(), ':p:h')
+  let g:temppathpattern = substitute(escape(b:temppath, '\\/.*$^~[]'), '[\\/]*$', '[\\\\/]\\+\\S\\+','')
+  let g:sometempfile = fnamemodify(b:temppath .'/some_temp_file.tmp', ':p')
+
+  runtime ale_linters/cs/mcsc.vim
+
+After:
+  Restore
+  unlet! b:ale_cs_mcsc_options
+  unlet! g:ale_cs_mcsc_source
+  unlet! g:ale_cs_mcsc_assembly_path
+  unlet! g:ale_cs_mcsc_assemblies
+  unlet! g:temppath
+  unlet! g:temppathpattern
+  unlte! g:sometempfile
+  call ale#linter#Reset()
+
+Execute(Check for proper default command):
+
+  let b:command = ale_linters#cs#mcsc#GetCommand(bufnr(''))
+  let b:command = substitute(b:command, '-out:' . g:temppathpattern, '-out:' . g:sometempfile, '')
+  let b:command = substitute(b:command, '\s\+', ' ', 'g')
+
+  AssertEqual
+  \   b:command,
+  \   'cd ".";mcs -unsafe -out:' . g:sometempfile . ' -t:module -recurse:"*.cs"'
+
+Execute(The options should be be used in the command):
+
+  let g:ale_cs_mcsc_options = '-pkg:dotnet'
+
+  let b:command = ale_linters#cs#mcsc#GetCommand(bufnr(''))
+  let b:command = substitute(b:command, '-out:' . g:temppathpattern, '-out:' . g:sometempfile, '')
```

Comment:
`sometmpefile` will cause problems here with `\` in paths on Windows. I recommend using something simpler like `'-out:TEMP_FILE` and testing for that. You're replacing it anyway.